sdk/cosmosdb/cosmos/test/public/functional/item.spec.ts

```diff
@@ -228,471 +527,1273 @@ describe("Item CRUD", function (this: Suite) {
+describe("item read retries", async function () {
+  it("retries on 429", async function () {
+    const client = new CosmosClient({ key: masterKey, endpoint });
+    const { resource: db } = await client.databases.create({
+      id: `small db ${Math.random() * 1000}`,
     });
+    const containerResponse = await client
+      .database(db.id)
+      .containers.create({ id: `small container ${Math.random() * 1000}`, throughput: 400 });
+    const container = containerResponse.container;
+    await container.items.create({ id: "readme" });
+    const arr = new Array(400);
+    const promises = [];
+    for (let i = 0; i < arr.length; i++) {
+      promises.push(container.item("readme").read());
+    }
+    const resp = await Promise.all(promises);
+    assert.equal(resp[0].statusCode, 200);
   });
-  describe("with v1 container", function () {
-    let container: Container;
-    let readItemId: string;
-    let replaceItemId: string;
-    let deleteItemId: string;
-    before(async function () {
-      container = await getTestContainer("bulk container", undefined, {
-        partitionKey: {
-          paths: ["/key"],
-          version: undefined,
-        },
-        throughput: 25100,
+});
+
+describe("bulk/batch item operations", async function () {
+  describe("test bulk operations", async function () {
+    describe("Check size based splitting of batches", function () {
+      let container: Container;
+      before(async function () {
+        container = await getTestContainer("bulk container", undefined, {
+          partitionKey: {
+            paths: ["/key"],
+            version: undefined,
+          },
+          throughput: 400,
+        });
       });
-      readItemId = addEntropy("item1");
-      await container.items.create({
-        id: readItemId,
-        key: "A",
-        class: "2010",
+      after(async () => {
+        await container.database.delete();
       });
-      deleteItemId = addEntropy("item2");
-      await container.items.create({
-        id: deleteItemId,
-        key: "A",
-        class: "2010",
+      it("Check case when cumulative size of all operations is less than threshold", async function () {
+        const operations: OperationInput[] = [...Array(10).keys()].map(
+          () =>
+            ({
+              ...generateOperationOfSize(100, { partitionKey: "key_value" }, { key: "key_value" }),
+            } as any)
+        );
+        const response = await container.items.bulk(operations);
+        // Create
+        response.forEach((res, index) =>
+          assert.strictEqual(res.statusCode, 201, `Status should be 201 for operation ${index}`)
+        );
       });
-      replaceItemId = addEntropy("item3");
-      await container.items.create({
-        id: replaceItemId,
-        key: 5,
-        class: "2010",
+      it("Check case when cumulative size of all operations is greater than threshold", async function () {
```

Comment:
maybe we can club both this and one before test together as both of them have same assert statement